### DISCUSSION TITLE
Inquires regarding the design of the IndentedTextWriter class

### DISCUSSION THREAD

**Discussion by FilipToth** [dotnet/runtime#54078] [General] [answered]
Hi, all! I just have a couple of questions about the design choices made when designing the `IndentedTextWriter` class.

## The namespace
Why did the runtime teams decide on putting this into `System.CodeDom.Compiler`? I understand it's a third-level namespace, which is good for these not so crucial APIs. But I don't feel like putting this into `System.CodeDom.Compiler` serves its purpose. Something like `System.IO` would be more appropriate, In my opinion.

## Where is it?
I was browsing the runtime's source and I couldn't find the `IndentedTextWriter` class in the `System.CodeDom.Compiler` namespace. I did find `ExposedTabStringIndentedTextWriter`, but that inherits from `IndentedTextWriter` so, still not the correct class.

## The problems with changing the namespace
- That would be a breaking change and we all don't like those especially for something minor like this.
- Polluting the System.IO namespace with this class is something I'm not sure we want to do at this stage.

So, what did I miss?

PS: I didn't want to open an issue for something as minor as this, and the review/runtime teams are already very busy as it is, so I'll just put it here.

**Comment by huoyaoyuan** [comment, marked as answer]
> Why did the runtime teams decide on putting this into System.CodeDom.Compiler?

Because it was introduced and designed with CodeDom

> Where is it?

Search with https://source.dot.net , and you will find it in CoreLib.

The whole CodeDom product should be considered legacy after roslyn came out. Just don't touch it.

**Reply by FilipToth** [reply]
I see, are there any plans on having an updated `IndentedTextWriter` class or something similar? I know it doesn't solve a big problem, but still would be nice to have.

**Reply by huoyaoyuan** [reply]
I'm not seeing any. `System.Text.Json` doesn't even rely on `TextWriter` because it's not performant enough.

**Reply by FilipToth** [reply]
Alright, thank you for the informative answer.
